### PR TITLE
Fix unicode support of regexp escape method #2282

### LINKED ISSUE #2282: Page slugs are no longer created correctly
**Describe the bug**  
https://forum.getkirby.com/t/page-slug-not-generated-correctly/16186

### LINKED ISSUE #2293: nothing to repeat error in tags field with 3 asterisks
**Describe the bug**  
When you type `***` into a tags field an error will be thrown: `nothing to repeat error in tags field`.

**To Reproduce**  
1. Edit a page with a tags field
2. insert three asterisks into it, one by one

**Expected behavior**  
No result or showing all results. I tried to use `*` as a placeholder value.

**Screenshots**  
This is what the field looks like afterwards:
grafik

**Kirby Version**  
3.3.0

**Console output**  
SyntaxError: "nothing to repeat"
    search http://localhost/kirby/media/panel/593ed86d9e989b97a31fa84858ccd5a0/js/app.js:1
    type http://localhost/kirby/media/panel/593ed86d9e989b97a31fa84858ccd5a0/js/app.js:1
    input http://localhost/kirby/media/panel/593ed86d9e989b97a31fa84858ccd5a0/js/app.js:1
    oe http://localhost/kirby/media/panel/593ed86d9e989b97a31fa84858ccd5a0/js/vendor.js:41
    n http://localhost/kirby/media/panel/593ed86d9e989b97a31fa84858ccd5a0/js/vendor.js:41
    _wrapper http://localhost/kirby/media/panel/593ed86d9e989b97a31fa84858ccd5a0/js/vendor.js:41

**Additional context**  
multilang site

### PR DESCRIPTION
**Pull request by afbora**
## Describe the PR

In the `replace` method, the `u` flag was not rendered properly, converted to the `new RegExp()` method with corrected pattern. 

**Note:** I have to remove `|` pipe char from escape pattern to work with `Str:$ascii` chars. Do you have a solution for this?

## Related issues

- Fixes #2282 
- Fixes #2293 

## Ready?

- [x] Added unit tests for fixed bug/feature
- [ ] Passing all unit tests
- [ ] Fixed code style issues with CS fixer and `composer fix`
- [ ] Added in-code documentation (if neeed)

### CODE REVIEW

**Comment by afbora** [comment]
Panel dist files to test: [panel.zip](https://github.com/getkirby/kirby/files/3814235/panel.zip)

File: panel/src/helpers/regex.js [hunk 1]
```diff
@@ -1 +1 @@
-RegExp.escape = s => s.replace(/[\p{L}]|[-\/\\^$*+?.()|[\]{}]+/u, '\\$&');
+RegExp.escape = s => s.replace(new RegExp("[\\p{L}][-/\\\\^$*+?.()|[\\]{}]", "gu"), '\\$&');
```
Comment (nilshoerrmann) [inline panel/src/helpers/regex.js]: Is it intentional that the first pipe is gone here compared to the original expression?

**Comment by afbora** [comment]
Slugs properly working but asteriks issue occured on #2293

File: panel/src/helpers/regex.js (hunk 1 above)
Comment (afbora) [reply panel/src/helpers/regex.js]: Absolutely you right, it needs to stay 👍

**Comment by bastianallgeier** [comment]
I added a few unit tests

**Review by bastianallgeier — APPROVED** [review]

**Pull request merged**